### PR TITLE
[Buddy] fix leaking go-routines in event-handler watcher

### PR DESCRIPTION
**Pull request by Joerger**
Buddy PR for https://github.com/gravitational/teleport-plugins/pull/963 with edits to resolve remaining comments.

### CODE REVIEW

File: event-handler/teleport_events_watcher.go [hunk 1]
```diff
@@ -20,6 +20,10 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
```
Comment (zmb3) [inline event-handler/teleport_events_watcher.go:25]: Wrong context import?

**Comment by Joerger** [comment]
I gave this some more thought and realized this one line fix is not a viable solution. This change allows the event handler to drop incomplete pages of events and move on to the next page. Instead, we need to update the event handler to only spawn a single goroutine at a time, instead of spawning new goroutines for new pages when we are still trying to finish a known page. Drafting for now as I won't have time to make this change immediately.

**Comment by Joerger** [comment]
I gave another pass to this fix, and found it works after all. I've updated the tests to give better coverage to the event collection logic and cover the edge case I was concerned about.

**Review by zmb3 — APPROVED** [review]
